### ISSUE TITLE
Update Loki proto

### ISSUE THREAD

**Issue by shuiyisong** [GreptimeTeam/greptimedb#5122]
### What type of enhancement is this?

API improvement

### What does the enhancement do?

https://github.com/GreptimeTeam/greptimedb/blob/c0f498b00c263998a521d6c4f36f63aacf4f875e/src/servers/Cargo.toml#L69 

We use [loki-api](https://crates.io/crates/loki-api) to introduce Loki rs proto. definition However it's out of date now. We might need to update the proto in one way or another.

### Implementation challenges

Possible solutions:
1. fork the `loki-api` repo and update the proto file.
2. copy and maintain the proto file directly in our codebase like [vector](https://github.com/vectordotdev/vector/tree/master/lib/loki-logproto/proto) does

**Comment by ozewr** [comment]
hello i'd like to work on this issue

**Comment by shuiyisong** [comment]
Sure. I also mentioned this in https://github.com/GreptimeTeam/greptimedb/issues/5287, No.4 in the TODO list.
I was thinking about maintaining it in https://github.com/GreptimeTeam/greptime-proto but has not determined yet.
We can have a discussion in this thread if you've come up with a plan.

Have fun!

**Comment by ozewr** [comment]
Do we need to update loki-api to this version? https://github.com/grafana/loki/blob/main/pkg/logproto/logproto.proto

**Comment by shuiyisong** [comment]
> Do we need to update loki-api to this version? https://github.com/grafana/loki/blob/main/pkg/logproto/logproto.proto

Yes. We also need to update https://github.com/grafana/loki/blob/main/pkg/push/push.proto
You can check vector's approach for reference: https://github.com/vectordotdev/vector/tree/master/lib/loki-logproto

**Comment by ozewr** [comment]
It seems that only need to place the new proto files into https://github.com/GreptimeTeam/greptime-proto/tree/main/proto and then expose them in https://github.com/GreptimeTeam/greptime-proto/blob/main/src/lib.rs  ？ After updating the proto, there will be some minor changes, such as adding the `structured_metadata` and `parsed` fields to `EntryAdapter`. Is this okay?

**Comment by shuiyisong** [comment]
Hi @ozewr . Sorry for the delay.

After discussion, we believe it's better to maintain a separate repo for holding Loki-related proto files(much like our greptime-proto repo) so that we can release it as an independent crate for better reusability.

Please check out the new repo: https://github.com/GreptimeTeam/loki-proto. You can place proto files there(through PR requests) and then import the crate into the GreptimeDB repo replacing the outdated `loki-api`.

Have fun coding! :)

**Comment by shuiyisong** [comment]
@ozewr Hi, I've noticed you haven't made a PR to the database to replace the `loki-api` crate, just checking in to ask if you are still working on this?

**Comment by ozewr** [comment]
> [@ozewr](https://github.com/ozewr) Hi, I've noticed you haven't made a PR to the database to replace the `loki-api` crate, just checking in to ask if you are still working on this?

Sorry, I’ve been quite busy recently, which caused me to forget about this issue. I will complete it in the next few days.
